### ISSUE TITLE
How to datalad update and merge new branch created in remote?

### ISSUE THREAD

**Issue by dorianps** [datalad/datalad#7447]
### What is the problem?

I could not find this information somewhere. What is the best way to merge a new branch created by a collaborator in a remote?

I can probably do that with git commands, but I was wondering if there is a specific way to handle the situation with datalad.

Local:
```
$ git branch
  GoodImages
  git-annex
* master
  synced/master
```

Remote:
```
]$ git branch
* ABC
  git-annex
  master
```

### DataLad information

datalad 0.17.6

**Comment by jsheunis** [comment]
Hi @dorianps, sorry for missing this and not getting back to you sooner.

The "datalad way" wouldn't be much different from doing it with git commands. 

If you do a `datalad update`, datalad will fetch updates from the remote, but do nothing else (i.e. no merging). You can specify how exactly an update procedure should run with the `--how` option, e.g. `--how fetch` (the default) or `--how merge`. In the latter case, the merge happens from the remote branch with a name that matches the current branch or from a specified sibling with a configured branch that is different from the current local branch (this requires configuration though, meaning extra lines of code).

If you don't bother with siblings, you will first have to checkout the specific new branch locally (or rather, create a new local branch that follows the new remote branch). Alternatively you could merge directly from the new remote branch into your local clone's default branch.

In summary:

this

```
git fetch
git merge myremote/newbranch
```

is the same as this
```
datalad update
git merge myremote/newbranch  # or git merge remotes/myremote/newbranch
```

**Comment by mih** [comment]
I am closing this issue. Feel free to reopen, if the question is not fully answered.
